playbooks/cleanup_pubcloud.yml

```diff
@@ -1,15 +1,14 @@
 ---
-- hosts: localhost
+- hosts: job_nodes
```

Comment:
Doing it this way will require a successful connection to the target host, even though it runs on tasks on there it still has to connect.

I don't know of a local connection will work in this case.

Also, what happens if the host never got created and therefore the connection attempt fails causing the playbook to fail.

Reply:
Currently, if an instance fails to build the cleanup will fail so that isn't much of a change. I don't think its entirely unreasonable either. No instance, so it can't be cleaned up. 

We could possibly improve that situation by making cleanup test for an inventory file and deciding that no inventory file means no instances so don't try to cleanup. 

One case we discussed in person is cleaning up instances that have been shutdown as a result of image creation. That case will fail. As discussed that mechnaism will likely change in  the near future when we implement node pool, so it may not be worth fixing. However we will need to ensure that the periodic clenaup job regex will catch those instances.